src/EnergyPlus/EarthTube.cc

```diff
@@ -587,23 +858,27 @@ void CalcEarthTube(EnergyPlusData &state)
         thisZoneHB.EAMFLxHumRat = 0.0;
         thisEarthTube.FanPower = 0.0;
 
-        // Skip this if the zone is below the minimum temperature limit
-        if (state.dataZoneTempPredictorCorrector->zoneHeatBalance(NZ).MAT < thisEarthTube.MinTemperature) continue;
-        // Skip this if the zone is above the maximum temperature limit
-        if (state.dataZoneTempPredictorCorrector->zoneHeatBalance(NZ).MAT > thisEarthTube.MaxTemperature) continue;
-        // Skip if below the temperature difference limit
-        if (std::abs(state.dataZoneTempPredictorCorrector->zoneHeatBalance(NZ).MAT - state.dataEnvrn->OutDryBulbTemp) < thisEarthTube.DelTemperature)
-            continue;
-
-        AirDensity =
-            Psychrometrics::PsyRhoAirFnPbTdbW(state, state.dataEnvrn->OutBaroPress, state.dataEnvrn->OutDryBulbTemp, state.dataEnvrn->OutHumRat);
+        bool tempShutDown = false;
+        // Don't simulate for Basic Solution if the zone is below the minimum temperature limit
+        if (thisZoneHB.MAT < thisEarthTube.MinTemperature) tempShutDown = true;
+        // Don't simulate for Basic Solution if the zone is above the maximum temperature limit
+        if (thisZoneHB.MAT > thisEarthTube.MaxTemperature) tempShutDown = true;
```

Comment:
Can this section just be:

```
bool tempShutDown = thisZoneHB.MAT > earthTube.MaxTemperature || thisZoneHB.MAT < earthTube.MinTemperature;
```

Reply:
Oooo, I guess I hadn't thought about such an assignment being possible and was trying to maintain as much similarity to the old code.  But this looks slick (or as the kids say these days "sick"), so I'll implement this also.  This "old dog" (me) needs to learn a few new tricks...

Reply:
Yeah, in Python it could just use an interval comparison :scream: 

```
tempShutDown = not (earthTube.MinTemperature < thisZoneHB.MAT < earthTube.MaxTemperature)
```

Reply:
I had such grand hopes last school year.  My now 16 yo was still doing homeschooling and so we were working through a Python intro course together.  Unfortunately I had too much to do to keep up with him.  Anyway, that sounds cool!